src/bundler.ts

```diff
@@ -15,10 +22,38 @@ export async function isGemOutdated(): Promise<boolean> {
   return false;
 }
 
+export async function addGem(): Promise<void> {
+  await execInPath("bundle add ruby-lsp --group=development");
+  await execInPath("bundle install");
+}
+
 export async function updateGem(): Promise<{ stdout: string; stderr: string }> {
   return execInPath("bundle update ruby-lsp");
 }
 
+export async function bundleCheck(): Promise<boolean> {
+  let bundlerCheck: string;
+
+  try {
+    bundlerCheck = (await execInPath("bundle check")).stdout;
+  } catch {
+    bundlerCheck = "";
+  }
+
+  if (bundlerCheck.includes("The Gemfile's dependencies are satisfied")) {
```

Comment:
Is it possible to use the exit code rather than checking the text?

Reply:
Just need to add `error: Error` to the object type in the `Promise` that is being returned, and check if `error` is `null` or not.

So: `: Promise<{ stdout: string; stderr: string }>` becomes `: Promise<{ stdout: string; stderr: string, error: Error }>` and `bundlerCheck` becomes:

```js
let bundlerCheck = (await execInPath("bundle check").error
```
and this line becomes:
```suggestion
  return !bundlerCheck;
```

Reply:
Ooops, I am sorry, what I wrote above isn't exactly correct, since I assumed too much about how `util.promisify` works.

According to the [docs](https://nodejs.org/docs/latest-v12.x/api/child_process.html#child_process_child_process_exec_command_options_callback):
> If this method is invoked as its [util.promisify()](https://nodejs.org/docs/latest-v12.x/api/util.html#util_util_promisify_original)ed version, it returns a `Promise` for an `Object` with `stdout` and `stderr` properties. The returned `ChildProcess` instance is attached to the `Promise` as a `child` property. In case of an error (including any error resulting in an exit code other than `0`), a rejected promise is returned, with the same `error` object given in the callback, but with two additional properties `stdout` and `stderr`.

So there is no need to change the return type of `execInPath` (and it would be wrong to change it anyway), the fact that an exception is raised from `await` and that we are catching it means that the exec failed. That should be good enough to decide if `bundle check` succeeded or not.